observ/watcher.py

```diff
@@ -137,6 +137,8 @@ class Watcher(Generic[T]):
         "_number_of_callback_args",
         "__weakref__",
     )
+    on_created: Optional[Callable[[Watcher[T]], None]] = None
```

Comment:
Because we use `from __future__ import annotations` we can actually use:
`Callable[[Watcher[T]], None] | None`. And also replace `Union` with `|`. 
It just came to mind, and it's out of scope for this PR. I'll open a separate issue.

Reply:
I've actually updated #114  with this idea.